### PR TITLE
Improved manual 

### PR DESCRIPTION
**Pull request by martingrossmann**
# Description

* Added a table with links to former versions of Testerra manual
* Added important notes to test controller chapter
* Removed duplicate browser cap chapter
* Some minor other stuff

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mbeuthan — APPROVED** [review]

File: docs/src/docs/browsercaps/common-browser-caps.adoc [hunk 1]
```diff
@@ -23,19 +23,27 @@ WEB_DRIVER_MANAGER.setGlobalCapability(FirefoxOptions.FIREFOX_OPTIONS, options);
...
 ====
 
+= Request capabilities
 
-== Request capabilities
-
-Some WebDriverRequests support setting capabilities, like the `DesktopWebDriverRequest`.
+A request defined capability means using a Webdriver request like `DesktopWebDriverRequest`.
 
+.Set capabilities to a DesktopWebDriverRequest object
```
Comment (Zsar) [inline docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Leading dot does not seem to do anything (and also does not get printed by Github).

File: docs/src/docs/browsercaps/common-browser-caps.adoc [hunk 2]
```diff
@@ -23,19 +23,27 @@ WEB_DRIVER_MANAGER.setGlobalCapability(FirefoxOptions.FIREFOX_OPTIONS, options);
...
-WebDriver webDriver = WEB_DRIVER_MANAGER.getWebDriver(myRequest);
+// Start your session with the DesktopWebDriverRequest object
+WebDriver driver = WebDriverManager.getWebDriver(request);
 ----
 
-== User agent configuration
+[NOTE]
+=====
+Have a look into <<Browser specific knowledge>> for specific browser options. +
+Find some more details for `DesktopWebDriverRequest` at <<_request_configuration>>.
```
Comment (Zsar) [inline docs/src/docs/browsercaps/common-browser-caps.adoc:43]: Anchor `_request_configuration` does not exist? (And also did not seem to exist previously, so unsure whence it should point.)

File: docs/src/docs/browsercaps/common-browser-caps.adoc [hunk 3]
```diff
@@ -23,19 +23,27 @@ WEB_DRIVER_MANAGER.setGlobalCapability(FirefoxOptions.FIREFOX_OPTIONS, options);
...
 myRequest.getDesiredCapabilities().put(CapabilityType.ACCEPT_INSECURE_CERTS, true);
-WebDriver webDriver = WEB_DRIVER_MANAGER.getWebDriver(myRequest);
+// Start your session with the DesktopWebDriverRequest object
+WebDriver driver = WebDriverManager.getWebDriver(request);
 ----
 
-== User agent configuration
+[NOTE]
+=====
+Have a look into <<Browser specific knowledge>> for specific browser options. +
```
Comment (Zsar) [inline docs/src/docs/browsercaps/common-browser-caps.adoc:42]: Anchor `Browser specific knowledge` does not exist? (And also did not seem to exist previously, so unsure whence it should point.)

**Review by Zsar — CHANGES_REQUESTED** [review]
(Interrupted after first two files. Going to continue later today.)

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 1 above)
Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: The leading dot means that the following text is formated as a description for the following code snippet.

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 3 above)
Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:42]: The links points to chapter in `browser-specific-knowledge.adoc`.

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 2 above)
Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:43]: This string is another format for `Request configuration` and it points to `webdrivermanager-config.adoc`

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 1 above)
Comment (Zsar) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Mmh. Does not seem to have a tangible effect [when viewed in GitHub](https://github.com/telekom/testerra/blob/docs/add-links-to-tt-version-docs/docs/src/docs/browsercaps/common-browser-caps.adoc). Which platform is this supposed to be rendered correctly on?

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 2 above)
Comment (Zsar) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:43]: GitHub turns this into a relative anchor on this file though: The generated link is `https://github.com/telekom/testerra/blob/docs/add-links-to-tt-version-docs/docs/src/docs/browsercaps/common-browser-caps.adoc#_request_configuration`.

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 3 above)
Comment (Zsar) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:42]: It points to `https://github.com/telekom/testerra/blob/docs/add-links-to-tt-version-docs/docs/src/docs/browsercaps/common-browser-caps.adoc#Browser%20specific%20knowledge` though.

File: docs/src/docs/index.adoc [hunk 4]
```diff
@@ -42,7 +42,8 @@ include::best-practices.adoc[leveloffset=+1]
 include::migration.adoc[leveloffset=+1]
 include::known-issues.adoc[leveloffset=+1]
 include::properties.adoc[leveloffset=+1]
-include::architecture.adoc[leveloffset+-1]
+include::architecture.adoc[leveloffset=+1]
```
Comment (Zsar) [inline docs/src/docs/index.adoc:45]: Formatting in GitHub seems to be broken: All generated links are on the same line.

**Review by Zsar — CHANGES_REQUESTED** [review]

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 1 above)
Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Please build it by your own with `gradle asciidoctor` and view the HTML result. You will find it under `docs/build/docs/html5/index.html`

Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Ähm... yes, I forgot: #112

Comment (Zsar) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Mmh. Works in IntelliJ though. Oh well.

File: docs/src/docs/browsercaps/common-browser-caps.adoc [hunk 5]
```diff
@@ -23,19 +23,27 @@ WEB_DRIVER_MANAGER.setGlobalCapability(FirefoxOptions.FIREFOX_OPTIONS, options);
 
 ====
 
+= Request capabilities
 
-== Request capabilities
-
-Some WebDriverRequests support setting capabilities, like the `DesktopWebDriverRequest`.
+A request defined capability means using a Webdriver request like `DesktopWebDriverRequest`.
```
Comment (Zsar) [inline docs/src/docs/browsercaps/common-browser-caps.adoc:28]: "means" seems to be the wrong verb here? Or possibly the object does not fit?
If I turn this around as a test: "Using a Webdriver request [...] means a request defined capability." does not read particularly sensible.

The old text seems more fitting to me, but it presumably was replaced for a reason?

**Review by Zsar — CHANGES_REQUESTED** [review]

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 1 above)
Comment (Zsar) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Had to adjust my `gradle-wrapper.properties` to use `gradle-6.9.2-bin.zip` (is 7.4), then it worked.

Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:30]: Yeah! :-)

File: docs/src/docs/execution/test-controller.adoc [hunk 6]
```diff
@@ -16,6 +16,9 @@ CONTROL.collectAssertions(() -> {
 });
 ----
 
+IMPORTANT: Assertions can only collected by using <<_regular_assertions>> or `expect()`/`assertThat()` of UiElements. +
```
[the remark this answers is not in the document]
Comment (martingrossmann) [reply docs/src/docs/execution/test-controller.adoc:19]: Fixed

File: docs/src/docs/doc-versions.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+|https://docs.testerra.io/testerra/1.12/index.html
+
+|1.13
+|https://docs.testerra.io/testerra/1.13/index.html
+
+|2.0
+|https://docs.testerra.io/testerra/2.0/index.html
+|===
+
+You will find detailed release notes at https://github.com/telekom/testerra.
```
[the remark this answers is not in the document]
Comment (martingrossmann) [reply docs/src/docs/doc-versions.adoc:27]: Updated the Github link.
And yes, we have no changelog.md. From my point the generated release notes should be enough.

[the remark this answers is not in the document]
Comment (Zsar) [reply docs/src/docs/doc-versions.adoc:27]: Yes, of course. It's just something I looked for as explanation, why the link would point to the code view, root folder.

Very well.

File: docs/src/docs/browsercaps/common-browser-caps.adoc (hunk 5 above)
Comment (martingrossmann) [reply docs/src/docs/browsercaps/common-browser-caps.adoc:28]: Hm... yes... The text is the result of merging the content of deleted `local-caps.adoc`. I reverted the change.

**Review by Zsar — APPROVED** [review]

**Review by gogro — APPROVED** [review]

**Pull request merged**